chain/epoch_manager/src/lib.rs

```diff
@@ -142,6 +142,63 @@ impl EpochManager {
         Ok(epoch_manager)
     }
 
+    /// Only used in mock network
+    /// Copy the necessary epoch info related to `block_hash` from `source_epoch_manager` to
+    /// the current epoch manager.
+    /// Note that this function doesn't copy info stored in EpochInfoAggregator, so `block_hash` must be
+    /// the last block in an epoch in order for the epoch manager to work properly after this function
+    /// is called
+    #[cfg(feature = "mock_network")]
+    pub fn copy_epoch_info_as_of_block(
+        &mut self,
+        block_hash: &CryptoHash,
+        epoch_id: &EpochId,
+        next_epoch_id: &EpochId,
+        source_epoch_manager: &mut EpochManager,
+    ) -> Result<(), EpochError> {
+        let mut store_update = self.store.store_update();
+        self.save_epoch_info(
+            &mut store_update,
+            epoch_id,
+            source_epoch_manager.get_epoch_info(epoch_id)?.clone(),
+        )?;
+        // save next epoch info too
+        self.save_epoch_info(
+            &mut store_update,
+            next_epoch_id,
+            source_epoch_manager.get_epoch_info(next_epoch_id)?.clone(),
+        )?;
+        // save next next epoch info if the block is the last block
```

Comment:
wait - but according to the comment - this should always be true, right? (it should always be  last block)

Reply:
Yes, I might change the code in the future to get rid of the requirement that it has to be the last block of the epoch